src/science/mathematics/linear_algebra/vector-norm.md

```diff
@@ -1,48 +1,58 @@
 ## Vector Norm
 [![ndarray-badge]][ndarray]
 
-This recipe demonstrates use of the [`Array1`] type, [`dot`], [`map`], and
-[`scalar_sum`] in computing the [l1] and [l2] norms of a given vector. The l2
-norm calculation is the simpler of the two, as it is the square root of the dot
-product of a vector with itself, shown in the function `l2_norm`. The l1 norm,
-shown in the function `l1_norm` is computed by mapping the elements of the input
-vector to their absolute value, and then calling `scalar_sum`, which reduces the
-elements via summation.
-
-Note that both `l1_norm` and `l2_norm` take the [`Array1`] type. This recipe
+This recipe demonstrates use of the [`Array1`] type, [`ArrayView1`] type,
+[`fold`] method, and [`dot`] method in computing the [l1] and [l2] norms of a
+given vector. The l2 norm calculation is the simpler of the two, as it is the
+square root of the dot product of a vector with itself, shown in the function
+`l2_norm`. The l1 norm, shown in the function `l1_norm`, is computed by a `fold`
+operation that sums the absolute values of the elements. (This could also be
+performed with `x.mapv(f64::abs).scalar_sum()`, but that would allocate a new
+array for the result of the `mapv`.)
+
+Note that both `l1_norm` and `l2_norm` take the [`ArrayView1`] type. This recipe
 considers vector norms, so the norm functions only need to accept one
-dimensional arrays (hence [`Array1`]).
+dimensional views (hence [`ArrayView1`]). While the functions could take a
```

Comment:
This level of detail is helpful.  Typically we shoot for smaller descriptions, but I'm going to keep it as is.

Reply:
Yeah, there might be a better place for this explanation, or maybe we should add it to the `ndarray` docs instead and leave it out of the cookbook. I thought it would be useful for newcomers, though, since one of the first questions when writing functions with `ndarray` types is what parameter type is best. I'll follow your guidance on whether to keep/remove this.